### PR TITLE
add some necessary config to boot up an xline cluster in benchmark

### PR DESCRIPTION
**Pull request by Phoenix500526**
We set 100ms to simulate the latency on the WAN, but our retry timeout and rpc timeout is only 50ms. Therefore,we need to extent the rpc timeout and retry timeout

Please briefly answer these questions:

* what problem are you trying to solve? (or if there's no problem, what's the motivation for this change?)

* what changes does this pull request make?

* are there any non-obvious implications of these changes? (does it break compatibility with previous versions, etc)

### CODE REVIEW

File: .github/workflows/benchmark.yml [hunk 1]
```diff
@@ -1,7 +1,8 @@
 name: Benchmark
 
 on:
-  workflow_dispatch: {}
```
[the remark this answers is not in the document]
Comment (Phoenix500526) [reply .github/workflows/benchmark.yml:4]: Fixed

File: scripts/benchmark.sh [hunk 2]
```diff
@@ -44,14 +44,22 @@ benchmark_cmd() {
...
+    --client-wait-synced-timeout 10s \
+    --client-propose-timeout 5s \
+    --cmd-workers 16"
 
     if [ ${1} -eq 1 ]; then
         cmd="${cmd} --is-leader"
     fi
 
     docker exec -e RUST_LOG=debug -d node${1} ${cmd}
-    echo "command is: docker exec -e RUST_LOG=debug -d node${1} ${cmd}"
```
[the remark this answers is not in the document]
Comment (Phoenix500526) [reply scripts/benchmark.sh:54]: Fixed

**Comment by Phoenix500526** [comment]
> Shall we use [educe](https://crates.io/crates/educe) crate to generate default values.

I think there is no need to use `educe` to generate default values, at least not in this pr. 
There are two reasons why we have to write these methods:
1. We need to provide a fine-grained way to configure default values when serializing and deserializing. The `serde` attribute macro provides a specific field attribute, named `[default](https://serde.rs/container-attrs.html#default--path)`, to set the default value of a single field. 
2. Be it `wait_synced_timeout` or `retry_timeout`, all these timeout objects are essential `Duration`. So what we need is to provide different contextual default values for objects of the same type, which cannot be done with `educe`.

**Review by rogercloud — DISMISSED** [review]

File: scripts/benchmark.sh [hunk 3]
```diff
@@ -238,7 +262,7 @@ for server in "xline" "etcd"; do
                 printf ${FORMAT} ${QPS} ${Latency} ${Time} "${cmd}" >>${result_file}
 
             done
+            stop_all
```
Comment (themanforfree) [inline scripts/benchmark.sh:265]: restart containers takes a long time, I recommend cleaning up the cluster inside the container

File: scripts/benchmark.sh [hunk 4]
```diff
@@ -44,14 +44,22 @@ benchmark_cmd() {
 run_xline() {
     cmd="/usr/local/bin/xline \
     --name node${1} \
-    --members ${MEMBERS}"
+    --members ${MEMBERS} \
+    --storage-engine memory \
```
Comment (themanforfree) [inline scripts/benchmark.sh:48]: we should run benchmark with persistent storage

File: scripts/benchmark.sh (hunk 3 above)
Comment (Phoenix500526) [reply scripts/benchmark.sh]: Fixed

File: scripts/benchmark.sh (hunk 4 above)
Comment (Phoenix500526) [reply scripts/benchmark.sh]: Fixed

**Review by themanforfree — APPROVED** [review]

**Review by themanforfree — APPROVED** [review]

**Review by markcty — APPROVED** [review]

**Pull request merged**